addons/dockable_container/dockable_container.gd

```diff
@@ -84,6 +86,39 @@ func _input(event: InputEvent) -> void:
 		fit_child_in_rect(_drag_n_drop_panel, panel.get_child_rect())
 
 
+func convert_to_window(node: Node):
+	var old_layout = get_layout().clone()
+	remove_child(node)
```

Comment:
I don't know how how soft parenting works (i'll need some guidance :sweat_smile:)

Reply:
Don't worry!
So, "hard parenting" is when you make a node be an actual child of another one, which is the most usual.
"Soft parenting", on the other hand, just simulates the parenting relation of position/rotaion/scale/size without the node being an actual child of the parent in terms of the hierarchy tree.

Take for example the panels in Dockable Container. They ocupy the TabContainers' rectangle, having their position and size changed dynamically, but are still direct children of the Dockable Container. So they are **hard parented** to the Dockable Container (are actual children of it, in terms of the hierarchy tree), while **soft parented** to the TabContainers (move and resize based on it).

This way, the Dockable Container can focus on managing only it's own direct children, without worrying about any other nodes that are elsewhere in the hierarchy.
I implemented it this way to maintain the node hierarchy intact, so that using `get_node` to access panel nodes gives reliable results. It doesn't matter how deep panels are in nested splits, you can always rely on the node hierarchy being the same. Using `get_node("..")` in a panel will give you the Dockable Container itself, while `$SomeDockableContainer/Panel1` will give you the `Panel1` node regardless of the current layout.
That's the reasoning behind wanting to maintain the hierarchy, mostly to have consistent results with `get_node` regardless of the layout configuration.

It is possible to implement floating windows using soft parenting while also maintaining this information in the Layout resource, but the implementation will be quite more complex. The thing I said before about draw order can be worked out by reordering the nodes, which changes the hierarchy in terms of node indices, but without any reparenting.

What do you think about all of this?
I know that's a lot of information, sorry about that 😅

Reply:
> Don't worry! So, "hard parenting" is when you make a node be an actual child of another one, which is the most usual. "Soft parenting", on the other hand, just simulates the parenting relation of position/rotaion/scale/size without the node being an actual child of the parent in terms of the hierarchy tree.
> 
> Take for example the panels in Dockable Container. They ocupy the TabContainers' rectangle, having their position and size changed dynamically, but are still direct children of the Dockable Container. So they are **hard parented** to the Dockable Container (are actual children of it, in terms of the hierarchy tree), while **soft parented** to the TabContainers (move and resize based on it).
> 
> This way, the Dockable Container can focus on managing only it's own direct children, without worrying about any other nodes that are elsewhere in the hierarchy. I implemented it this way to maintain the node hierarchy intact, so that using `get_node` to access panel nodes gives reliable results. It doesn't matter how deep panels are in nested splits, you can always rely on the node hierarchy being the same. Using `get_node("..")` in a panel will give you the Dockable Container itself, while `$SomeDockableContainer/Panel1` will give you the `Panel1` node regardless of the current layout. That's the reasoning behind wanting to maintain the hierarchy, mostly to have consistent results with `get_node` regardless of the layout configuration.
> 
> It is possible to implement floating windows using soft parenting while also maintaining this information in the Layout resource, but the implementation will be quite more complex. The thing I said before about draw order can be worked out by reordering the nodes, which changes the hierarchy in terms of node indices, but without any reparenting.
> 
> What do you think about all of this? I know that's a lot of information, sorry about that sweat_smile

Hmmm, indeed it's a lot :exploding_head::sweat_smile: Honestly, i'm not that familiar with the Dockable container and implemented the fastest possible way to implement windows (it was originally for pixelorama but the idea was scrapped in hopes of it being implemented in this plugin for Godot 4 (the one i made is a Godot 3 variant))
So, this need not be a pull request :sweat_smile:. i just thought i should give you this code :grin: so that it would help you in making a more efficient window system